How-To-Setup-Internal-Package-Repository-Nexus.md

```diff
@@ -0,0 +1,241 @@
+# How To Setup An Internal Package Repository With Sonatype Nexus
+
+<!-- TOC -->
+
+- [Summary](#summary)
+- [Pre-Requisites](#pre-requisites)
+- [Install Sonatype Nexus](#install-sonatype-nexus)
+- [Create A Hosted Repository For Chocolatey Packages](#create-a-hosted-repository-for-chocolatey-packages)
+  - [Create User To Push Packages To The Repository](#create-user-to-push-packages-to-the-repository)
+  - [Obtain Repository API Key](#obtain-repository-api-key)
+  - [Push A Test Package](#push-a-test-package)
+  - [Install A Package](#install-a-package)
+- [Create a Proxy Repository For Chocolatey Community Repository](#create-a-proxy-repository-for-chocolatey-community-repository)
+  - [Install A Package](#install-a-package-1)
+  - [What Packages Is Your Proxy Repository Storing](#what-packages-is-your-proxy-repository-storing)
+- [Saving Your Repository Source In Chocolatey](#saving-your-repository-source-in-chocolatey)
+
+<!-- /TOC -->
+
+## Summary
+
+While the Chocolatey Community Repository is a great resource for packages, it is [not recommended for direct use by organizations](how-to-setup-internal-package-repository#organizational-requirements). But that doesn't mean you have to miss out on using the tens of thousands of packages that it contains. The solution is to setup an internal repository to host your own packages and we are going to discuss the use of [Sonatype Nexus](https://sonatype.com) for this task.
+
+There are two types of repositories you can setup:
+
+* Hosted Repository - this is where only the packages that have been pushed directly to the repository are available for use;
+* Proxy Repository - this is where you can use Nexus as a go-between for another repository, in this case the Chocolatey Community Repository. When you install or upgrade a package it is downloaded from that other repository and then stored (or cached) locally by Nexus. On each subsequent install / upgrade the Nexus package will be used. So you're only ever downloading from the other repository once.
+
+## Pre-Requisites
+
+Sonatype Nexus needs to be installed on a machine, whether that be a physical machine or a virtual machine. Follow these steps before doing so:
+
+1. Create a server with [Windows Server 2016](https://www.microsoft.com/en-us/evalcenter/evaluate-system-center) - this can be a virtual or physical server. Details on how to create a server are beyond the scope of this guide;
+1. [Install Chocolatey](https://chocolatey.org/install#installing-chocolatey);
+1. From the Chocolatey Community Repository [download](https://chocolatey.org/api/v2/package/putty/0.70) the [putty.install version 0.70](https://chocolatey.org/packages/putty.install/0.70) package - this will be used later during our tests;

```

Comment:
Please suggest PuTTY 0.71 here. 0.70 has a lot of bugs that are solved in 0.71:
https://www.chiark.greenend.org.uk/~sgtatham/putty/

Reply:
Version 0.70 has been suggested specifically as we will be using common packages across the examples (so we will build on this in other docs by using 0.71 and other packages). The goal is not to install and use but to install and see it working.